### ISSUE TITLE
Missing module: electron-publisher-undefined

### ISSUE THREAD

**Issue by ChiefOfGxBxL** [electron-userland/electron-builder#1628]
* **Version**: 18.5.1

* **Target**: Win x64
* **Node**: v.4.5.0
* **npm**: v5.0.3

The issue: I'm encountering a `Cannot find module 'electron-publisher-undefined'` error when I run `npm run dist`.

I'm setting up the automatic build-and-deploy using GitHub releases. I've specified a build configuration in my package.json:
```json
"build": {
    "appId": "com.github.blahblah",
    "publish": {
        "host": "github.com",
        "owner": "ChiefOfGxBxL",
        "private": true,
        "protocol": "https"
    }
  }
```

I've also set an environment variable `GH_TOKEN` with the proper permissions to `repo`, since this is a private repository.

As specified in the **quick getting started** guide, I put these two scripts:
```
"pack": "build --dir",
"dist": "build"
```

Now when I run `npm run dist`, I'm getting this error: 
```
Error: Cannot find module 'electron-publisher-undefined'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at requireProviderClass (C:\Users\Me\Desktop\blahblah\node_modules\electron-builder\src\publish\PublishManager.ts:344:14)
    at C:\Users\Me\Desktop\blahblah\node_modules\electron-builder\src\publish\PublishManager.ts:470:25
    at next (native)
```

Any recommendations on how to resolve this missing module? Is that supposed to read *electron-publisher-github* or something similar? I'm thinking since it is undefined a config variable is set wrong somewhere. Thanks!

**Comment by ChiefOfGxBxL** [comment]
Note that if I edit my `build.publish` to include the pair `provider: "github"` the error about the module not being found goes away... but no files are published on my repo's Releases

**Comment by develar** [comment]
Do not specify host, protocol and private

**Comment by develar** [comment]
Please set env DEBUG=electron-builder and attach log of the terminal output.

**Comment by ChiefOfGxBxL** [comment]
Hi develar, I've attached the debug output.

I also removed `host`, `protocol`, and `private`. My `GH_TOKEN` is still set.

[debug.txt](https://github.com/electron-userland/electron-builder/files/1056057/debug.txt)

**Comment by develar** [comment]
Is repo blahblah exists? Or it is private repo?

**Comment by ChiefOfGxBxL** [comment]
It is a private repo, although I plan on making it public in the near future

**Comment by ChiefOfGxBxL** [comment]
I resolved this issue. First, my end of node_modules was hitting the wrong endpoint for the GitHub api (I removed `/repos/` from the URLs in an effort to resolve the previous issue).

Still, `electron-publish\out\gitHubPublisher.js` complained that it couldn't find the module `mime` that it was trying to `require()` on line 47. Not sure if that's just an issue that affects me.

In any case, I'm where I need to be now. Thanks for the help! I donated $10 on Donorbox
